### PR TITLE
Support for agent-free metrics reporting

### PR DESCRIPTION
**Pull request by agavra**
I tested this with a local controller setup directly reporting to DD: https://us5.datadoghq.com/s/7d2b7ec4-e391-11ed-b891-da7ad0900005/iua-jdy-8wc

In a follow-up PR I'll modify `kafka-client-examples/simple-example` to run with agent-free metrics reporting.

### CODE REVIEW

File: kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/ResponsiveMetrics.java [hunk 1]
```diff
@@ -48,10 +48,12 @@ public class ResponsiveMetrics implements Closeable {
   private static final Pattern GLOBAL_THREAD_REGEX = Pattern.compile(".*-(GlobalStreamThread+)");
 
   private OrderedTagsSupplier orderedTagsSupplier;
+  private final OtelMetricsService otelService;
```
Comment (rodesai) [inline kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/ResponsiveMetrics.java:51]: can we keep this service decoupled from the metrics collection stuff? Ideally we won't bake in the fact that we're using any specific reporter. One suggestion is to have some way to listen on `ResponsiveMetrics` being closed and register the otel reporter as a listener. And then move the otel stuff to its own package.

File: kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/OtelMetricsService.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    onClose.run();
+  }
+
+  private static void buildFromUserRules(MetricConfiguration conf) {
+    RuleParser parserInstance = RuleParser.get();
+    final ClassLoader loader = OtelMetricsService.class.getClassLoader();
+
+    // TODO(agavra): instead of including otel-jmx.config.yaml as a resource we should
+    // fetch it from the Responsive controller on start-up
+    try (final var is = loader.getResourceAsStream("otel-jmx.config.yaml")) {
```
Comment (rodesai) [inline kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/OtelMetricsService.java:109]: this is a huge improvement 🥳

File: kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/OtelMetricsService.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      parserInstance.addMetricDefsTo(conf, is, "responsive builtin config");
+    } catch (Exception e) {
+      // yaml parsing errors are caught and logged inside addMetricDefsTo
+      // only file access related exceptions are expected here
+      JmxMetricInsight.getLogger().warning(e.toString());
```
Comment (rodesai) [inline kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/OtelMetricsService.java:114]: I think this should be a fatal error that we should throw the exception up for. Now that we're getting the config from a resource, this indicates something is seriously wrong with the jar we're shipping.

File: kafka-client/src/main/resources/otel-jmx.config.yaml [hunk 4]
```diff
@@ -0,0 +1,334 @@
+---
```
Comment (rodesai) [inline kafka-client/src/main/resources/otel-jmx.config.yaml:1]: this is the same as https://github.com/responsivedev/responsive-pub/blob/main/etc/otel-jmx.config.yaml right? Can you remove that file now?

**Review by rodesai — COMMENTED** [review]
Awesome! Left a few thoughts inlline

File: kafka-client/src/main/resources/otel-jmx.config.yaml (hunk 4 above)
Comment (agavra) [reply kafka-client/src/main/resources/otel-jmx.config.yaml:1]: Yes, I will remove it once I migrate all the examples to use the builtin reporter

File: kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/ResponsiveMetrics.java (hunk 1 above)
Comment (agavra) [reply kafka-client/src/main/java/dev/responsive/kafka/internal/metrics/ResponsiveMetrics.java:51]: I'll wrap it in a `MetricsExporterService` and move no-op to be it's own implementation of that. That way we don't add yet another series of callbacks, they're hard to keep track of.

**Pull request merged**